modules/edu.gemini.seqexec.model/src/main/scala/edu/gemini/seqexec/model/Model.scala

```diff
@@ -128,19 +157,84 @@ object Model {
       }
     )
     // Composite lens to change the sequence name of an event
-    val sequenceNameL: PTraversal[SeqexecEvent, SeqexecEvent, String, String] = sePrism composeLens seViewL composeLens sequencesQueueL composeTraversal eachL composeLens obsNameL
+    val sequenceNameL: Traversal[SeqexecEvent, ObservationName] =
+      sePrism         ^|->  // Events with model updates
+      seViewL         ^|->  // Find the sequence view
+      sequencesQueueL ^|->> // Find the queue
+      eachViewL       ^|->  // each sequence on the queue
+      obsNameL              // sequence's observation name
+
+    // Composite lens to find the step config
+    val sequenceConfigL: Traversal[SeqexecEvent, StepConfig] =
+      sePrism           ^|->  // Events with model updates
+      seViewL           ^|->  // Find the sequence view
+      sequencesQueueL   ^|->> // Find the queue
+      eachViewL         ^|->  // each sequence on the queue
+      obsStepsL         ^|->> // sequence steps
+      eachStepL         ^<-?  // each step
+      standardStepL     ^|->  // which is a standard step
+      stepConfigL             // configuration of the step
+
+    // Composite lens to find the target name on observation
+    val observeTargetNameL: Traversal[SeqexecEvent, TargetName] =
+      sequenceConfigL                                 ^|-?  // configuration of the step
+      configTargetNameL(SystemName.observe, "object")       // on the configuration find the target name
+
+    // Composite lens to find the target name on telescope
+    val telescopeTargetNameL: Traversal[SeqexecEvent, TargetName] =
+      sequenceConfigL                                 ^|-?  // configuration of the step
+      configTargetNameL(SystemName.telescope, "Base:name")       // on the configuration find the target name
 
     implicit val equal: Equal[SeqexecEvent] = Equal.equalA
   }
 
-  type SystemName = String
+  // The system name in ocs is a string but we can represent the important ones as an ADT
```

Comment:
Is this one of those cases of types in ocs that are passed as String and then converted to other types in ocs3?

Reply:
almost all in ocs is a string. This just makes it simpler to work with with some risk that we are missing a type